specs/lightclient/lightclient-fork.md

```diff
@@ -0,0 +1,83 @@
+# Ethereum 2.0 Light Client Support -- From Phase 0 to Light Client Patch
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+## Table of contents
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Configuration](#configuration)
+- [Fork to Light-client patch](#fork-to-light-client-patch)
+  - [Fork trigger](#fork-trigger)
+  - [Upgrading the state](#upgrading-the-state)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Introduction
+
+This document describes the process of moving from Phase 0 to Phase 1 of Ethereum 2.0.
+
+## Configuration
+
+Warning: this configuration is not definitive.
+
+| Name | Value |
+| - | - |
+| `LIGHTCLIENT_PATCH_FORK_VERSION` | `Version('0x01000000')` |
+| `LIGHTCLIENT_PATCH_FORK_SLOT` | `Slot(0)` **TBD** |
+
+## Fork to Light-client patch
+
+### Fork trigger
+
+TBD. Social consensus, along with state conditions such as epoch boundary, finality, deposits, active validator count, etc. may be part of the decision process to trigger the fork. For now we assume the condition will be triggered at slot `LIGHTCLIENT_PATCH_FORK_SLOT`, where `LIGHTCLIENT_PATCH_FORK_SLOT % SLOTS_PER_EPOCH == 0`.
+
+### Upgrading the state
+
+After `process_slots` of Phase 0 finishes, if `state.slot == LIGHTCLIENT_PATCH_FORK_SLOT`, an irregular state change is made to upgrade to light-client patch.
+
+```python
+def upgrade_to_lightclient_patch(pre: phase0.BeaconState) -> BeaconState:
+    epoch = get_current_epoch(pre)
+    post = BeaconState(
+        genesis_time=pre.genesis_time,
+        slot=pre.slot,
+        fork=Fork(
+            previous_version=pre.fork.current_version,
+            current_version=LIGHTCLIENT_PATCH_FORK_VERSION,
+            epoch=epoch,
+        ),
+        # History
+        latest_block_header=pre.latest_block_header,
```

Comment:
@djrtwo @vbuterin 

Since `sync_committee_bits` and `sync_committee_signature` fields are added to block header instead of block body in https://github.com/ethereum/eth2.0-specs/pull/2130/commits/3b7c02514b9c855acdddbe2918bd0e02eeb02c9b, it would be more tricky to handle the SSZ objects at the fork boundary.

1. The `latest_block_header` field here *should* be transformed from `phase0.BeaconBlockBody` to `lightclient_patch.BeaconBlockBody`.
2. However, the next slot (the first slot after the light client patch) will have a problem when caching the `previous_block_root = hash_tree_root(state.latest_block_header)` in `process_slot` function because the previous root should have been computed with phase 0 form at phase 0.

I think it makes more sense to have `sync_committee_bits` and `sync_committee_signature` in `BeaconBlockBody` although they are not "operations".

Reply:
Fixed in https://github.com/ethereum/eth2.0-specs/pull/2141/commits/2c86b433ae13a59196ebb10ff9997eacf1af2939